GitVersionCore/GitPreparer.cs

```diff
@@ -202,10 +202,11 @@ private static Reference GetLocalReference(Repository repository, string branchN
             return repository.Refs.FirstOrDefault(localRef => string.Equals(localRef.CanonicalName, targetBranchName));
         }
 
-        private static DirectReference GetRemoteReference(Repository repository, string branchName, string repositoryUrl)
+        private static DirectReference GetRemoteReference(Repository repository, string branchName, Authentication authentication)
         {
             var targetBranchName = branchName.GetCanonicalBranchName();
-            var remoteReferences = repository.Network.ListReferences(repositoryUrl);
+            var remote = repository.Network.Remotes.Single();
+            var remoteReferences = GitHelper.GetRemoteTipsUsingUsernamePasswordCredentials(repository, remote, authentication.Username, authentication.Password);
```

Comment:
I just logged https://github.com/libgit2/libgit2sharp/issues/1099 in order to ease this kind of use case.
This shouldn't need a remote to work. One should be able to do this over an anonymous url.